enterprise/frontend/src/metabase-enterprise/snippets/components/SnippetCollectionForm.tsx

```diff
@@ -112,6 +113,16 @@ function SnippetCollectionForm({
     [collection.id, isEditing, handleCreate, handleUpdate, onSave],
   );
 
+  const shouldDisableItem = useCallback(
+    (item: CollectionPickerItem) => {
+      return passedCollection.id
+        ? item.effective_location?.includes(String(passedCollection.id)) ||
+            passedCollection.id === item.id
```

Comment:
I think you'll need to split effective_location, right?  otherwise you'll disable stuff overly-aggressively like

passedCollection.id: `11`
effective_location: `1/113/8113`

Reply:
Fair point! I'll get that fixed

Reply:
Good catch. doing a split